### PR TITLE
#107

### LINKED ISSUE #107: null values and addConcreteSettersForOptional
I find that addConcreteSettersForOptional uses `Optional#of` instead of `Optional#ofNullable` quite surprising especially since Intellij IDEA by default doesn't warn against passing nulls to `javax.validation.constraints.NotNull` annotated parameters.
What is the reasoning behind this decision?

### PR DESCRIPTION
**Pull request by lpandzic**
Fixes #107

### CODE REVIEW

**Comment by Randgalt** [comment]
I will review soon - thanks

File: record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java [hunk 1]
```diff
@@ -993,12 +1017,26 @@ public MyRecordBuilder p(T p) {
...
         methodSpec.addJavadoc("Set a new value for the {@code $L} record component in the builder\n", component.name())
-                .addStatement("this.$L = $T.of($L)", component.name(), type.typeName(), component.name());
+                .addStatement(getOptionalStatement(type), component.name(), type.typeName(), component.name());
         addConstructorAnnotations(component, parameterSpecBuilder);
+        filterOutNotNullAnnotationOnParameter(parameterSpecBuilder);
         methodSpec.addStatement("return this").addParameter(parameterSpecBuilder.build());
         builder.addMethod(methodSpec.build());
     }
 
+    private void filterOutNotNullAnnotationOnParameter(ParameterSpec.Builder parameterSpecBuilder) {
```
Comment (Randgalt) [inline record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java:1027]: I feel that this portion should be in a different PR. Let's focus this PR solely on the `ofNullable`. wdyt?

**Comment by Randgalt** [comment]
I apologize it's taken me so long to look at this. I left a comment for you to reply to.

File: record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java (hunk 1 above)
Comment (lpandzic) [reply record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java]: I agree since the issue discussion has gone off on a tangent.
But the issue this tries to solve is when there's a `@NotNull Optional` field in record.
In that case current code without the method above would generate 

```java
setX(@NotNull X x) {
    Optional.ofNullable(x);
}
``` 

This isn't a problem in my company code base because we never validate builders but it will surprise some.
I've reverted the commit that adds this filter.

Comment (Randgalt) [reply record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java]: I want to handle the null issue finally and completely which I hope to do with #111 - people seem to have strong opinions on this :D

**Comment by Randgalt** [comment]
Please rebase the commits

**Comment by lpandzic** [comment]
> Please rebase the commits

done

**Comment by Randgalt** [comment]
> > Please rebase the commits
> 
> done

I still see 7 commits. Did you force push your squash?

**Comment by lpandzic** [comment]
I'm not sure I follow - my branch has 7 commits that are ahead of the master. I don't see any commits on master that are not present in my branch.

**Comment by Randgalt** [comment]
Please squash down to 1 commit.
